src/read.rs

```diff
@@ -228,6 +230,50 @@ impl<R: Read+io::Seek> ZipArchive<R>
         })
     }
 
+    /// Extract a Zip archive into a directory.
+    ///
+    /// Paths are sanitized so that they cannot escape the given directory.
+    ///
+    /// # Platform-specific behaviour
+    ///
+    /// On unix systems permissions from the zip file are preserved, if they exist.
+    pub fn extract(&mut self, directory: &Path) -> ZipResult<()> {
+        for i in 0..self.len() {
+            let mut file = self.by_index(i)?;
+            let filepath = file.sanitized_name();
+
+            // `sanitized_name` should return a relative path
+            // otherwise there's a risk of directory traversal attacks
+            assert!(filepath.is_relative());
```

Comment:
Does this protect against someone using a path like `../../../alpaca.png`?

Reply:
The [`sanitized_name`](https://docs.rs/zip/0.5.3/zip/read/struct.ZipFile.html#method.sanitized_name) function should do that. This is just an extra sanity check.